kratos/python_scripts/python_linear_solver_factory.py

```diff
@@ -168,3 +168,27 @@ def ConstructSolver(configuration):
         KM.Logger.PrintInfo("Linear-Solver-Factory",\
             "Constructing a regular (non-complex) linear-solver")
         return KM.LinearSolverFactory().Create(configuration)
+
+def CreateFastestAvailableDirectLinearSolver():
+    # Using a default linear solver (selecting the fastest one available)
+    if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
+        from KratosMultiphysics import EigenSolversApplication
+    elif kratos_utils.CheckIfApplicationsAvailable("ExternalSolversApplication"):
+        from KratosMultiphysics import ExternalSolversApplication
+
+    linear_solvers_by_speed = [
+        "pardiso_lu", # EigenSolversApplication (if compiled with Intel-support)
+        "sparse_lu",  # EigenSolversApplication
+        "pastix",     # ExternalSolversApplication (if Pastix is included in compilation)
```

Comment:
IMO pastix is or the first or the second faster solver (we must update the solver BTW :P )

Reply:
I doubt it is faster than pardiso
at least here everyone switched from pastix to pardiso since it became available

Reply:
Perhaps only in my ill-conditioned contact problem

Reply:
:D

Why again was it necessary to update it?
I remember there was an issue but I forgot the details

Reply:
Because right now we are using the legacy version. The last version is faster and better, and MPI compatible BTW, but refactors everything, so the adaptation must be done from scratch

Reply:
oh, that sound like quite some effort :/

I would be curious to try the MPI-version

Reply:
http://pastix.gforge.inria.fr/files/README-txt.html according to the last papers published we speak of 4x-6x faster

Reply:
impressive

we cannot include it permantently bcs of the license right?

Reply:
yep, cecil is a strange licence, is basically a LGPL by INRIA